### PR TITLE
fix: fix `run_macaron.sh` script to handle action arguments correctly

### LINKED ISSUE #460: `./run_macaron.sh analyze -h` does not work as expected
# Info

**Macaron version**: 0.4

# Issue

The `run_macaron.sh` script does not behave as expected when running `run_macaron.sh analyze -h`: instead of printing the help text for the `analyze` action, it prints the help text of the `macaron` command.

```
> ./scripts/release_scripts/run_macaron.sh -h
Setting default output directory to /home/<user>/dev/orcl/macaron-ws/repo-finder-refactor/output.
Running ghcr.io/oracle/macaron:latest
latest: Pulling from oracle/macaron
Digest: sha256:e980f4eee44d981210e8dc9d3eca9109220d44ef407adb011e2bc9c37527c736
Status: Image is up to date for ghcr.io/oracle/macaron:latest
usage: macaron [-h] [-V] [-v] [-o OUTPUT_DIR] [-dp DEFAULTS_PATH] [-lr LOCAL_REPOS_PATH] {analyze,dump-defaults,verify-policy} ...

positional arguments:
  {analyze,dump-defaults,verify-policy}
                        Run macaron <action> --help for help

options:
  -h, --help            show this help message and exit
  -V, --version         Show Macaron's version number and exit
  -v, --verbose         Run Macaron with more debug logs
  -o OUTPUT_DIR, --output-dir OUTPUT_DIR
                        The output destination path for Macaron
  -dp DEFAULTS_PATH, --defaults-path DEFAULTS_PATH
                        The path to the defaults configuration file.
  -lr LOCAL_REPOS_PATH, --local-repos-path LOCAL_REPOS_PATH
                        The directory where Macaron looks for already cloned repositories.
```

**Expected behavior**: the help text of `./run_macaron.sh analyze -h` should be similar to the help text of `macaron analyze -h`.

**Cause**: the `./run_macaron.sh` script always treats the `-h` flag as a help flag for the main `macaron` command, but not the help flag for any of the actions in Macaron (e.g. `analyze` or `verify-policy`).

### PR DESCRIPTION
**Pull request by nathanwn**
Closes #460.

# Summary

* Fix the `run_macaron.sh` script to handle action arguments correctly.
* Add a test script to check for the arguments that are actually passed to the `macaron` command in the Macaron container.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -284,6 +284,7 @@ integration-test:
 .PHONY: integration-test-docker
 integration-test-docker:
 	scripts/dev_scripts/integration_tests_docker.sh $(REPO_PATH) scripts/release_scripts/run_macaron.sh
+	scripts/dev_scripts/test_run_macaron_sh.py
```
Comment (tromai) [inline Makefile:287]:
```suggestion
        scripts/dev_scripts/test_run_macaron_sh.py
	scripts/dev_scripts/integration_tests_docker.sh $(REPO_PATH) scripts/release_scripts/run_macaron.sh
```
The `test_run_macaron_sh.py` is shorter, I think we should place it first to exit early when there are errors?

Comment (behnazh-w) [reply Makefile:287]: Instead of adding the run command here, can you please call this as a test from `integration_tests_docker.sh`?

Comment (behnazh-w) [inline Makefile:287]: Instead of adding the run command here, can you please call this as a test from `integration_tests_docker.sh`?

Comment (nathanwn) [reply Makefile]: I have moved the command as you suggested. See 87bfcfec300f31e73f0a62b479c1206b4ddaecf4.

Comment (nathanwn) [reply Makefile]: I have moved `test_run_macaron_sh.py` to the `integration_tests_docker.sh` script, as suggested by Behnaz. See 87bfcfec300f31e73f0a62b479c1206b4ddaecf4.

**Review by tromai — APPROVED** [review]
LGTM

**Review by behnazh-w — APPROVED** [review]

**Pull request merged**